routers/api/v1/repo/branch.go

```diff
@@ -182,6 +182,87 @@ func DeleteBranch(ctx *context.APIContext) {
 	ctx.Status(http.StatusNoContent)
 }
 
+// CreateRepoBranch creates a branch for a user's repository
+func CreateRepoBranch(ctx *context.APIContext, opt api.CreateBranchRepoOption) {
+	err := repo_module.CreateNewBranch(ctx.User, ctx.Repo.Repository, opt.OldBranchName, opt.BranchName)
```

Comment:
check if branch exist or make OldBranchName optional

Reply:
The CreateNewBranch function checks if the branch exist and return and error if the branch already exists.

In the error checking below, there is a use of ```if models.IsErrBranchAlreadyExists(err)```. Is this check enough or should we add more specific check ?

Reply:
I would haver preverded if you just send a commit, squash things will make reviews harder, since you dont exactly know what has schanged?

Reply:
Ok sorry, since the changes are not very big, I wanted to be the commit to be very close to the initial message of the issue.

Reply:
all pulls will be squash-merged at the end :)

Reply:
Oh ok I see, sorry for that then :)